analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -1286,4 +1286,26 @@ public void M(object o)
                 }
             }
             """, allowSemanticModel);
+
+    [DataTestMethod]
+    [DataRow("var d = [u:dateTimePointer]->Date;", false)]
+    [DataRow("var d = dateTimePointer->[u:Date];", false)]
+    [DataRow("var d = (*[u:dateTimePointer]).Date;", false)]
+    [DataRow("var d = (*dateTimePointer).[u:Date];", false)]
+    [DataRow("[t:Int32]* iPointer;", false)]
+    [DataRow("[u:System].Int32* iPointer;", true)]
+    [DataRow("System.[t:Int32]* iPointer;", false)]
+    [DataRow("[k:void]* voidPointer;", false)]
+    [DataRow("DateTime d = default; M(&[u:d]);", false)]
+    public void IdentifierToken_Unsafe_Pointers(string statement, bool allowSemanticModel) =>
```

Comment:
There are some additional test cases that come to mind:
 ```cs
int?* x1; // Pointer of a nullable value type ("?" shouldn't be part of the token, right?
Nullable<int>* x2; // Nullable is of type t, whereas int is of type k, both are recognized
int** x3; // ** is not included in the token int, which is of type k
```
And relative assignments:
```cs
_ = &x1; // x1 of type u
_ = (*(&x1))->Value; // x1 and Value of type u
_ = (*x1).Value; // x1 and Value of type u
```

There are also ways of mixing `@` and `&`, and we may want to make sure we only highlight the right thing:
```cs
var @int = (int*)null; // 2nd int and null of type k, 1st int of type u (?) with @ included
_ = *@int; // @int of type u (?), * not included, @ included
_ = &@int; // @int of type u (?), & not included, @ included
```

You can also similarly mix `[]` with `*` and `&`.

Hereafter is my understanding of one of the cases mentioned above (to be verified):
```cs
[DataRow("var [u:@int] = ([k:int]*)[k:null];_ = *[u:@int];", false)]
```

Would it make sense to cover (some of) them?

Reply:
I added a couple of tests. The `@` tests are a bit unrelated and I separated those. The `@` is just parsed as part of the identifier token, so it doesn't really change anything.